crates/router/src/connector/trustpay.rs

```diff
@@ -0,0 +1,637 @@
+        }
+        _ => {
+            let mut header = vec![(headers::CONTENT_TYPE.to_string(), content_type)];
+            let mut api_key = auth_header;
+            header.append(&mut api_key);
+            Ok(header)
+        }
+    }
+}
+
+impl api::Payment for Trustpay {}
+
+impl api::PreVerify for Trustpay {}
+impl ConnectorIntegration<api::Verify, types::VerifyRequestData, types::PaymentsResponseData>
+    for Trustpay
+{
+}
+
+impl api::PaymentVoid for Trustpay {}
+
+impl ConnectorIntegration<api::Void, types::PaymentsCancelData, types::PaymentsResponseData>
+    for Trustpay
+{
+}
+
+impl api::ConnectorAccessToken for Trustpay {}
+
+impl ConnectorIntegration<api::AccessTokenAuth, types::AccessTokenRequestData, types::AccessToken>
+    for Trustpay
+{
+    fn get_url(
+        &self,
+        _req: &types::RefreshTokenRouterData,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        Ok("https://aapi.trustpay.eu/api/oauth2/token".to_owned())
+    }
+
+    fn get_content_type(&self) -> &'static str {
+        "application/x-www-form-urlencoded"
+    }
+
+    fn get_headers(
+        &self,
+        req: &types::RefreshTokenRouterData,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        let auth = trustpay::TrustpayAuthType::try_from(&req.connector_auth_type)
+            .change_context(errors::ConnectorError::FailedToObtainAuthType)?;
+        let auth_value = format!(
+            "Basic {}",
+            consts::BASE64_ENGINE.encode(format!("{}:{}", auth.project_id, auth.secret_key))
+        );
+        Ok(vec![
+            (
+                headers::CONTENT_TYPE.to_string(),
+                types::RefreshTokenType::get_content_type(self).to_string(),
+            ),
+            (headers::AUTHORIZATION.to_string(), auth_value),
+        ])
+    }
+
+    fn get_request_body(
+        &self,
+        req: &types::RefreshTokenRouterData,
+    ) -> CustomResult<Option<String>, errors::ConnectorError> {
+        let trustpay_req =
+            utils::Encode::<trustpay::TrustpayAuthUpdateRequest>::convert_and_url_encode(req)
+                .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+        Ok(Some(trustpay_req))
+    }
+
+    fn build_request(
+        &self,
+        req: &types::RefreshTokenRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Option<services::Request>, errors::ConnectorError> {
+        let req = Some(
+            services::RequestBuilder::new()
+                .method(services::Method::Post)
+                .headers(types::RefreshTokenType::get_headers(self, req, connectors)?)
+                .url(&types::RefreshTokenType::get_url(self, req, connectors)?)
+                .body(types::RefreshTokenType::get_request_body(self, req)?)
+                .build(),
+        );
+        logger::debug!(trustpay_access_token_request=?req);
```

Comment:
already things are logged in common function

Reply:
Done